app-modules/knowledge-base/resources/views/filament/pages/badges.blade.php

```diff
@@ -33,7 +33,32 @@
 --}}
 
 <div class="flex gap-2">
+    @if ($getRecord()->status)
+        <x-filament::badge>
+            {{ $getRecord()?->status?->name ?? '' }}
+        </x-filament::badge>
+    @endif
     <x-filament::badge>
-        {{ $getRecord()?->status?->name ?? '' }}
+        {{ !empty($getRecord()->public) ? 'Public' : 'Internal' }}
     </x-filament::badge>
+    @if (!empty($getRecord()->public) && !empty($getRecord()->category_id))
+        <x-filament::badge icon="heroicon-m-clipboard">
+            <button
+                type="button"
+                x-data
+                x-on:click="
+        window.navigator.clipboard.writeText(@js(route('portal.knowledge-management.show') . '/categories/' . $getRecord()->category_id . '/articles/' . $getRecord()->id)).then(() => {
+            $tooltip('Copied!', {
+                theme: $store.theme,
+                timeout: 2000,
+            })
+        }).catch(err => {
+            console.error('Failed to copy text: ', err);
+        })
+    "
+            >
+                Copy
+            </button>
+        </x-filament::badge>
```

Comment:
Let's have the copy icon and the click event happen actually on the public badge itself rather than a being a completely separate badge.

You can have the icon be on the right side in the badge as well by using `icon-position="after"`: https://filamentphp.com/docs/3.x/support/blade-components/badge#adding-an-icon-to-a-badge

Reply:
Done.